tests/soak-tests/src/test/java/org/apache/activemq/artemis/tests/soak/owleak/OWLeakTest.java

```diff
@@ -0,0 +1,242 @@
+            });
+         }
+
+
+         service.execute(() -> {
+
+            try {
+               ConnectionFactory factory = CFUtil.createConnectionFactory(protocol, "tcp://localhost:61616");
+               Connection connection = factory.createConnection();
+               Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+               MessageConsumer consumer = session.createConsumer(session.createQueue(QUEUE_NAME));
+               connection.start();
+
+               for (int i = 0; i < NUMBER_OF_MESSAGES * PRODUCERS; i++) {
+
+                  TextMessage message = (TextMessage) consumer.receive(100_000);
+                  Assert.assertNotNull(message);
+                  logger.info("Received {} messages , total of {}", i, (NUMBER_OF_MESSAGES * PRODUCERS));
+                  semaphore.release();
+               }
+
+            } catch (Throwable e) {
+               errors.incrementAndGet();
+               logger.warn(e.getMessage(), e);
+            } finally {
+               running.set(false);
+               latch.countDown();
+            }
+         });
+
+         service.execute(() -> {
+            ConnectionFactory factory = CFUtil.createConnectionFactory("core", "tcp://localhost:61616");
+            try {
+               Connection connection = factory.createConnection();
+               Session session = connection.createSession(true, Session.SESSION_TRANSACTED);
+               MessageProducer producer = session.createProducer(session.createQueue("fastQueue"));
+               MessageConsumer consumer = session.createConsumer(session.createQueue("fastQueue"));
+               connection.start();
+               long msg = 0;
+               char[] msgStr = new char[1024];
+               String buffer = new String(msgStr);
+               Arrays.fill(msgStr, 'a');
+               while (running.get()) {
+                  TextMessage message = session.createTextMessage(buffer);
+                  producer.send(message);
+                  if (++msg % 10000L == 0L) {
+                     logger.info("Sent and receive {} fast messages", msg);
+                  }
+
+                  if (msg > 5000L) {
+                     message = (TextMessage) consumer.receive(10000);
+                     Assert.assertNotNull(message);
+                  }
+
+                  if (msg % 100L == 0L) {
+                     session.commit();
+                  }
+               }
+               session.commit();
+               producer.close();
+               consumer.close();
+               session.close();
+               connection.close();
+            } catch (Exception e) {
+               errors.incrementAndGet();
+               e.printStackTrace();
+               logger.warn(e.getMessage(), e);
+            } finally {
+               latch.countDown();
+               running.set(false);
+            }
+         });
+
+
+         Assert.assertTrue(latch.await(10, TimeUnit.DAYS));
```

Comment:
This is excessive, nothing should be able to silently hold the test suite for minutes, let alone days.

Reply:
It was a test thing I forgot to rollback.  Thanks.